### PR TITLE
Update NominatedNodeName for pipelined task

### PR DESCRIPTION
**Pull request by bibibox**
To fix issue #3499

### CODE REVIEW

**Comment by lowang-bh** [comment]
This pr can be part of solution for feature [record the norminated node for pod and consider it first in next cycle #3491](https://github.com/volcano-sh/volcano/issues/3491)

**Comment by bibibox** [comment]
> This pr can be part of solution for feature [record the norminated node for pod and consider it first in next cycle #3491](https://github.com/volcano-sh/volcano/issues/3491)

yes, there may be some overlap, and this pr is mainly to solve this issue #3499

File: pkg/scheduler/cache/cache.go [hunk 1]
```diff
@@ -992,14 +992,26 @@ func (sc *SchedulerCache) taskUnschedulable(task *schedulingapi.TaskInfo, reason
...
+	updateNomiNode := podNominatedNodeNameNeedUpdate(&pod.Status, nominatedNodeName)
+
+	if updateCond || updateNomiNode {
 		pod = pod.DeepCopy()
 
+		if updateCond && podutil.UpdatePodCondition(&pod.Status, condition) {
+			klog.V(3).Infof("Updating pod condition for %s/%s to (%s==%s)", pod.Namespace, pod.Name, condition.Type, condition.Status)
+		}
+
+		if updateNomiNode {
```
Comment (Monokaix) [inline pkg/scheduler/cache/cache.go:1005]: Better add a comment here to explain why this is added？

File: pkg/scheduler/api/job_info.go [hunk 2]
```diff
@@ -669,19 +669,19 @@ func (ji *JobInfo) TaskSchedulingReason(tid TaskID) (reason string, msg string)
 	case Allocated:
 		// Pod is schedulable
 		msg = fmt.Sprintf("Pod %s/%s can possibly be assigned to %s", taskInfo.Namespace, taskInfo.Name, ctx.NodeName)
-		return PodReasonSchedulable, msg
+		return PodReasonSchedulable, msg, ""
```
Comment (Monokaix) [inline pkg/scheduler/api/job_info.go:672]: `Allocated` is also a unschedulable case, we should also check whether nominatedNode is needed, maybe we can refer to kube -scheduler to deep dive.

Comment (bibibox) [reply pkg/scheduler/api/job_info.go:672]: the autoscaler will check both condition status and reason whether a pod is unschedulable or not:
image
so allocated will not be thought as unschedulable case to autoscaler

File: pkg/scheduler/cache/cache.go (hunk 1 above)
Comment (bibibox) [reply pkg/scheduler/cache/cache.go:1007]: done

File: pkg/scheduler/api/job_info.go (hunk 2 above)
Comment (Monokaix) [reply pkg/scheduler/api/job_info.go:672]: Ok

**Review by william-wang — APPROVED** [review]
/approve

**Pull request merged**